aiida/engine/processes/workchains/utils.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+"""Utilities for `WorkChain` implementations."""
+from collections import namedtuple
+from functools import wraps
+
+from ..exit_code import ExitCode
+
+__all__ = ('ProcessHandler', 'ProcessHandlerReport', 'register_process_handler')
+
+ProcessHandler = namedtuple('ProcessHandler', 'method priority exit_codes')
```

Comment:
Maybe replace `method` with `process_class`? I am not sure what exactly you wanted to say here, but it is definitely confusing.

Reply:
The `method` argument will reference the actual function that was decorated with the `registered_process_handler` decorator. I decided to call it a method instead of function since once it has been decorated and turned into this tuple, the decorated function actually has become a method of the target process class.

Reply:
After considering renaming the `method` element to `handler` I realized that this really makes the code in `BaseRestartWorkChain.inspect_process` difficult to understand. In the loop over registered handlers, the call `handler.method(self, node)` would become `handler.handler(self, node)` which is less clear. Since the `ProcessHandler` tuple is anyway only for internal use and does not concern the user, I have left it this way. I _did_ clarify the docstring of the tuple with regards to the meaning of the `method` tuple element.

Reply:
looks good to me.